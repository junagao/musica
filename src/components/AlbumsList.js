import React from 'react';
import PropTypes from 'prop-types';

import AlbumItem from './AlbumItem';

const AlbumsList = ({ albums }) => {
  return (
    <div>
      <ul className="album-list">
        {albums.map((album) => {
          const {
            collectionId,
            artistName,
            artworkUrl100,
            collectionName,
          } = album;
          return (
            <AlbumItem
              key={collectionId}
              artwork={artworkUrl100}
              artist={artistName}
              album={collectionName}
            />
          );
        })}
      </ul>
    </div>
  );
};

AlbumsList.propTypes = {
  albums: PropTypes.arrayOf(
    PropTypes.shape({
      key: PropTypes.number,
      artwork: PropTypes.string,
      artist: PropTypes.string,
      album: PropTypes.string,
    }),
  ).isRequired,
};

export default AlbumsList;
