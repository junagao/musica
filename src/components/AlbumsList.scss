@import '../variables';

.albums-list-container {
  padding: 2rem 0rem 0rem 0rem;
}

.albums-list-header {
  display: grid;
  grid-template-columns: 1fr 1fr;
}

h2 {
  color: $color-h2;
  font-size: 1.5rem;
}

.view-type-container {
  justify-self: end;
}

.view-btn {
  color: $color-lighter;
  margin-left: 0.75rem;
  font-size: 1.5rem;
  cursor: pointer;
  outline: none;
  border: 0;
  text-decoration: none;
  -webkit-appearance: none;
  -moz-appearance: none;
  transition: all ease-out 0.3s;
  &:hover,
  &:focus {
    color: $color-darker;
    transition: all ease-in 0.3s;
  }
}

.albums-list {
  display: grid;
  grid-column-gap: 2rem;
  grid-row-gap: 1.5rem;
  padding: 1rem 0rem 2rem 0rem;
  justify-content: space-between;
  @media (min-width: $tabletP) {
    grid-column-gap: 1rem;
  }
  @media (min-width: $laptop) {
    grid-column-gap: 4rem;
  }
  @media (min-width: $desktop) {
    grid-column-gap: 6rem;
  }
  @media (min-width: $desktopHD) {
    grid-column-gap: 6rem;
  }
}

.grid {
  grid-template-columns: repeat(auto-fill, 150px);
  animation: fadeIn 1s;
  @media (min-width: $desktopHD) {
    grid-template-columns: repeat(auto-fill, 175px);
  }
}

.list {
  grid-template-rows: 1fr 1fr;
  grid-auto-rows: row;
  animation: fadeIn 1s;
}

@keyframes fadeIn {
  from {
    opacity: 0;
  }
  to {
    opacity: 1;
  }
}
