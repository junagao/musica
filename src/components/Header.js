import React from 'react';

import './Header.scss';

const Header = () => (
  <header>
    <h1>
      <a href="/">musica</a>
    </h1>
  </header>
);

export default Header;
