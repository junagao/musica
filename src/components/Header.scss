@import '../variables';

h1 {
  color: $color-h1;
  font-size: 4.5rem;
  font-weight: $font-weight-h1;
  @media (min-width: $mobileL) {
    font-size: 5rem;
  }
  @media (min-width: $tabletP) {
    font-size: 5.5rem;
  }
}
