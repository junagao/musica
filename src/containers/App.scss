@import '../variables';

.container {
  margin: 0 auto;
  max-width: 22rem;
  overflow: hidden;
  padding: 0.5rem;
  @media (min-width: $mobileL) {
    max-width: 38rem;
  }
  @media (min-width: $tabletP) {
    max-width: 44rem;
  }
  @media (min-width: $tabletL) {
    max-width: 56rem;
  }
  @media (min-width: $laptop) {
    max-width: 64rem;
  }
  @media (min-width: $desktop) {
    max-width: 72rem;
  }
  @media (min-width: $desktopHD) {
    max-width: 92rem;
  }
}

.error-message {
  color: #e53e3e;
  font-size: 1.25rem;
  padding: 2rem 0rem;
}
