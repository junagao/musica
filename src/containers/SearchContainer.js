import React from 'react';
import PropTypes from 'prop-types';

import SearchButton from '../components/SearchButton';
import SearchInput from '../components/SearchInput';

class SearchContainer extends React.Component {
  constructor() {
    super();
    this.state = {
      searchTerm: '',
    };
  }

  onSearchTermChange = (e) => {
    this.setState({ searchTerm: e.target.value });
  };

  handleSubmit = (e) => {
    const { onSearch } = this.props;
    const { searchTerm } = this.state;
    e.preventDefault();
    onSearch(searchTerm);
    e.currentTarget.reset();
  };

  render() {
    return (
      <form className="search-form" onSubmit={this.handleSubmit}>
        <SearchInput onSearchTermChange={this.onSearchTermChange} />
        <SearchButton />
      </form>
    );
  }
}

SearchContainer.propTypes = {
  onSearch: PropTypes.func.isRequired,
};

export default SearchContainer;
