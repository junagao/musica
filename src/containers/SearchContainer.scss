.search-form {
  display: flex;
  align-items: center;
  margin-right: 0.75rem;
  height: 54px;
}
