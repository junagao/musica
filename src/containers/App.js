import React from 'react';
import PropTypes from 'prop-types';
import { hot } from 'react-hot-loader/root';
import { connect } from 'react-redux';
import { getAlbums as getAlbumsAction } from '../actions/albums';

import Header from '../components/Header';
import SearchContainer from './SearchContainer';
import AlbumsList from '../components/AlbumsList';
import NoAlbum from '../components/NoAlbum';
import ErrorMessage from '../components/ErrorMessage';

import './App.scss';

export class App extends React.Component {
  handleSearch = (term) => {
    const { getAlbums } = this.props;

    getAlbums(term);
  };

  handleChangeView = (type) => {
    const albumListContainer = document.getElementsByClassName(
      'albums-list',
    )[0];
    const albumItens = document.getElementsByClassName('album-item');

    if (type === 'list') {
      albumListContainer.classList.replace('grid', 'list');
      albumItens.forEach((i) => i.classList.replace('grid', 'list'));
    } else {
      albumListContainer.classList.replace('list', 'grid');
      albumItens.forEach((i) => i.classList.replace('list', 'grid'));
    }
  };

  render() {
    const { data, loading, error } = this.props;

    return (
      <div className="container">
        <Header />
        <SearchContainer onSearch={this.handleSearch} loading={loading} />
        {error && <ErrorMessage error={error} />}
        {!error && data.length ? (
          <AlbumsList albums={data} onChangeView={this.handleChangeView} />
        ) : (
          <NoAlbum />
        )}
      </div>
    );
  }
}

App.propTypes = {
  data: PropTypes.arrayOf(
    PropTypes.shape({
      key: PropTypes.number,
      artwork: PropTypes.string,
      artist: PropTypes.string,
      album: PropTypes.string,
    }),
  ),
  error: PropTypes.string,
  loading: PropTypes.bool.isRequired,
  getAlbums: PropTypes.func,
};

App.defaultProps = {
  data: [],
  error: '',
  getAlbums: () => {},
};

const mapStateToProps = (state) => ({
  data: state.albums.data,
  loading: state.albums.loading,
  error: state.albums.error.message,
});

const mapDispatchToProps = {
  getAlbums: getAlbumsAction,
};

export const connectedApp = hot(
  connect(mapStateToProps, mapDispatchToProps)(App),
);
