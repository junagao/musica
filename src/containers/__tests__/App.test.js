import React from 'react';
import { mount } from 'enzyme';
import { Provider } from 'react-redux';

import store from 'store';
import { App } from 'containers/App';
import Header from 'components/Header';
import SearchContainer from 'containers/SearchContainer';
import AlbumsList from 'components/AlbumsList';

describe('App', () => {
  let wrapper;

  beforeEach(() => {
    const mockAlbums = [
      {
        collectionId: 793928184,
        artistName: 'Tycho',
        collectionName: 'Awake',
        artworkUrl100:
          'https://is4-ssl.mzstatic.com/image/thumb/Music/v4/b9/76/bc/b976bc5d-2c25-d79d-4057-67bc47b4ab6d/source/100x100bb.jpg',
      },
      {
        collectionId: 1461523692,
        artistName: 'Tycho',
        collectionName: 'Weather',
        artworkUrl100:
          'https://is1-ssl.mzstatic.com/image/thumb/Music123/v4/a9/bf/0c/a9bf0c63-fb1d-b3ee-315f-40d829f0159c/source/100x100bb.jpg',
      },
    ];
    wrapper = mount(
      <Provider store={store}>
        <App data={mockAlbums} loading={false} error="" />
      </Provider>,
    );
  });

  afterEach(() => {
    wrapper.unmount();
  });

  it('should render correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('renders one div with className `container`', () => {
    expect(
      wrapper
        .find('div')
        .at(0)
        .hasClass('container'),
    ).toBeTruthy();
  });

  it('shows a Header', () => {
    expect(wrapper.find(Header).length).toEqual(1);
  });

  it('shows a SearchContainer', () => {
    expect(wrapper.find(SearchContainer).length).toEqual(1);
  });

  it('shows a ErrorMessage if theres an error', () => {
    wrapper.setProps({ error: 'ERROR' });
    expect(wrapper.find(SearchContainer).length).toEqual(1);
  });

  it('shows a AlbumsList if theres data', () => {
    expect(wrapper.find(AlbumsList).length).toEqual(1);
  });
});
