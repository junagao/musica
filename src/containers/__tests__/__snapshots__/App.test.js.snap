// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should render correctly 1`] = `
<Provider
  store={
    Object {
      "dispatch": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
      Symbol(observable): [Function],
    }
  }
>
  <App
    albumsPerPage={20}
    changeCurrentPage={[Function]}
    currentPage={1}
    data={
      Array [
        Object {
          "artistName": "Tycho",
          "artworkUrl100": "https://is4-ssl.mzstatic.com/image/thumb/Music/v4/b9/76/bc/b976bc5d-2c25-d79d-4057-67bc47b4ab6d/source/100x100bb.jpg",
          "collectionId": 793928184,
          "collectionName": "Awake",
        },
        Object {
          "artistName": "Tycho",
          "artworkUrl100": "https://is1-ssl.mzstatic.com/image/thumb/Music123/v4/a9/bf/0c/a9bf0c63-fb1d-b3ee-315f-40d829f0159c/source/100x100bb.jpg",
          "collectionId": 1461523692,
          "collectionName": "Weather",
        },
      ]
    }
    error=""
    getAlbums={[Function]}
    loading={false}
    noResults={false}
  >
    <div
      className="container"
    >
      <Header>
        <header>
          <h1>
            <a
              href="/"
            >
              musica
            </a>
          </h1>
        </header>
      </Header>
      <SearchContainer
        loading={false}
        onSearch={[Function]}
        s={true}
      >
        <form
          className="search-form"
          onSubmit={[Function]}
        >
          <SearchInput
            onSearchTermChange={[Function]}
          >
            <label
              className="search-field"
              htmlFor="search"
            >
              <input
                className="search-input"
                name="search"
                onChange={[Function]}
                placeholder="Search"
                type="search"
              />
            </label>
          </SearchInput>
          <SearchButton>
            <button
              className="search-button"
              type="submit"
            >
              <IoIosSearch>
                <IconBase
                  attr={
                    Object {
                      "viewBox": "0 0 512 512",
                    }
                  }
                >
                  <svg
                    fill="currentColor"
                    height="1em"
                    stroke="currentColor"
                    strokeWidth="0"
                    style={
                      Object {
                        "color": undefined,
                      }
                    }
                    viewBox="0 0 512 512"
                    width="1em"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M443.5 420.2L336.7 312.4c20.9-26.2 33.5-59.4 33.5-95.5 0-84.5-68.5-153-153.1-153S64 132.5 64 217s68.5 153 153.1 153c36.6 0 70.1-12.8 96.5-34.2l106.1 107.1c3.2 3.4 7.6 5.1 11.9 5.1 4.1 0 8.2-1.5 11.3-4.5 6.6-6.3 6.8-16.7.6-23.3zm-226.4-83.1c-32.1 0-62.3-12.5-85-35.2-22.7-22.7-35.2-52.9-35.2-84.9 0-32.1 12.5-62.3 35.2-84.9 22.7-22.7 52.9-35.2 85-35.2s62.3 12.5 85 35.2c22.7 22.7 35.2 52.9 35.2 84.9 0 32.1-12.5 62.3-35.2 84.9-22.7 22.7-52.9 35.2-85 35.2z"
                      key="0"
                    />
                  </svg>
                </IconBase>
              </IoIosSearch>
            </button>
          </SearchButton>
        </form>
      </SearchContainer>
      <AlbumsList
        albums={
          Array [
            Object {
              "artistName": "Tycho",
              "artworkUrl100": "https://is4-ssl.mzstatic.com/image/thumb/Music/v4/b9/76/bc/b976bc5d-2c25-d79d-4057-67bc47b4ab6d/source/100x100bb.jpg",
              "collectionId": 793928184,
              "collectionName": "Awake",
            },
            Object {
              "artistName": "Tycho",
              "artworkUrl100": "https://is1-ssl.mzstatic.com/image/thumb/Music123/v4/a9/bf/0c/a9bf0c63-fb1d-b3ee-315f-40d829f0159c/source/100x100bb.jpg",
              "collectionId": 1461523692,
              "collectionName": "Weather",
            },
          ]
        }
        noResults={false}
        onChangeView={[Function]}
      >
        <div
          className="albums-list-container"
        >
          <div
            className="albums-list-header"
          >
            <h2>
              Albums
            </h2>
            <div
              className="view-type-container"
            >
              <ViewTypeButton
                btState=""
                icon={<FaThList />}
                onChangeView={[Function]}
                viewType="list"
              >
                <button
                  className="view-btn list-btn "
                  onClick={[Function]}
                  type="button"
                >
                  <FaThList>
                    <IconBase
                      attr={
                        Object {
                          "viewBox": "0 0 512 512",
                        }
                      }
                    >
                      <svg
                        fill="currentColor"
                        height="1em"
                        stroke="currentColor"
                        strokeWidth="0"
                        style={
                          Object {
                            "color": undefined,
                          }
                        }
                        viewBox="0 0 512 512"
                        width="1em"
                        xmlns="http://www.w3.org/2000/svg"
                      >
                        <path
                          d="M149.333 216v80c0 13.255-10.745 24-24 24H24c-13.255 0-24-10.745-24-24v-80c0-13.255 10.745-24 24-24h101.333c13.255 0 24 10.745 24 24zM0 376v80c0 13.255 10.745 24 24 24h101.333c13.255 0 24-10.745 24-24v-80c0-13.255-10.745-24-24-24H24c-13.255 0-24 10.745-24 24zM125.333 32H24C10.745 32 0 42.745 0 56v80c0 13.255 10.745 24 24 24h101.333c13.255 0 24-10.745 24-24V56c0-13.255-10.745-24-24-24zm80 448H488c13.255 0 24-10.745 24-24v-80c0-13.255-10.745-24-24-24H205.333c-13.255 0-24 10.745-24 24v80c0 13.255 10.745 24 24 24zm-24-424v80c0 13.255 10.745 24 24 24H488c13.255 0 24-10.745 24-24V56c0-13.255-10.745-24-24-24H205.333c-13.255 0-24 10.745-24 24zm24 264H488c13.255 0 24-10.745 24-24v-80c0-13.255-10.745-24-24-24H205.333c-13.255 0-24 10.745-24 24v80c0 13.255 10.745 24 24 24z"
                          key="0"
                        />
                      </svg>
                    </IconBase>
                  </FaThList>
                </button>
              </ViewTypeButton>
              <ViewTypeButton
                btState="active"
                icon={<FaThLarge />}
                onChangeView={[Function]}
                viewType="grid"
              >
                <button
                  className="view-btn grid-btn active"
                  onClick={[Function]}
                  type="button"
                >
                  <FaThLarge>
                    <IconBase
                      attr={
                        Object {
                          "viewBox": "0 0 512 512",
                        }
                      }
                    >
                      <svg
                        fill="currentColor"
                        height="1em"
                        stroke="currentColor"
                        strokeWidth="0"
                        style={
                          Object {
                            "color": undefined,
                          }
                        }
                        viewBox="0 0 512 512"
                        width="1em"
                        xmlns="http://www.w3.org/2000/svg"
                      >
                        <path
                          d="M296 32h192c13.255 0 24 10.745 24 24v160c0 13.255-10.745 24-24 24H296c-13.255 0-24-10.745-24-24V56c0-13.255 10.745-24 24-24zm-80 0H24C10.745 32 0 42.745 0 56v160c0 13.255 10.745 24 24 24h192c13.255 0 24-10.745 24-24V56c0-13.255-10.745-24-24-24zM0 296v160c0 13.255 10.745 24 24 24h192c13.255 0 24-10.745 24-24V296c0-13.255-10.745-24-24-24H24c-13.255 0-24 10.745-24 24zm296 184h192c13.255 0 24-10.745 24-24V296c0-13.255-10.745-24-24-24H296c-13.255 0-24 10.745-24 24v160c0 13.255 10.745 24 24 24z"
                          key="0"
                        />
                      </svg>
                    </IconBase>
                  </FaThLarge>
                </button>
              </ViewTypeButton>
            </div>
          </div>
          <ul
            className="albums-list grid"
          >
            <AlbumItem
              album="Awake"
              artist="Tycho"
              artwork="https://is4-ssl.mzstatic.com/image/thumb/Music/v4/b9/76/bc/b976bc5d-2c25-d79d-4057-67bc47b4ab6d/source/100x100bb.jpg"
              key="793928184"
            >
              <li
                className="album-item grid"
              >
                <img
                  alt=""
                  src="https://is4-ssl.mzstatic.com/image/thumb/Music/v4/b9/76/bc/b976bc5d-2c25-d79d-4057-67bc47b4ab6d/source/200x200bb.jpg"
                />
                <div
                  className="album-details"
                >
                  <p
                    className="album-title"
                  >
                    Awake
                  </p>
                  <p
                    className="album-artist"
                  >
                    Tycho
                  </p>
                </div>
              </li>
            </AlbumItem>
            <AlbumItem
              album="Weather"
              artist="Tycho"
              artwork="https://is1-ssl.mzstatic.com/image/thumb/Music123/v4/a9/bf/0c/a9bf0c63-fb1d-b3ee-315f-40d829f0159c/source/100x100bb.jpg"
              key="1461523692"
            >
              <li
                className="album-item grid"
              >
                <img
                  alt=""
                  src="https://is1-ssl.mzstatic.com/image/thumb/Music123/v4/a9/bf/0c/a9bf0c63-fb1d-b3ee-315f-40d829f0159c/source/200x200bb.jpg"
                />
                <div
                  className="album-details"
                >
                  <p
                    className="album-title"
                  >
                    Weather
                  </p>
                  <p
                    className="album-artist"
                  >
                    Tycho
                  </p>
                </div>
              </li>
            </AlbumItem>
          </ul>
        </div>
      </AlbumsList>
      <Pagination
        albumsPerPage={20}
        onChangePage={[Function]}
        totalAlbums={2}
      >
        <div>
          <nav>
            <ul
              className="pagination"
            >
              <li
                className="page-item"
                key="1"
              >
                <a
                  className="page-link"
                  href="#1"
                  onClick={[Function]}
                >
                  1
                </a>
              </li>
            </ul>
          </nav>
        </div>
      </Pagination>
    </div>
  </App>
</Provider>
`;
